dump checks that user has select privilege

jseldess opened the pull request:
Fixes #511

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/docs/512)
<!-- Reviewable:end -->

mjibson commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[back-up-and-restore-data.md, line 47 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/512#-KNfP6tkJf9Xz2VgW_3u:-KNfP6tkJf9Xz2VgW_3v:-95737213) ([raw file](https://github.com/cockroachdb/docs/blob/966b6728cc133a789080a42f8e0aa215f6d50e39/back-up-and-restore-data.md#L47)):*
> ```Markdown
> `--pretty` | Not valid for the `dump` command. This flag will eventually be removed.
> `--url` | The connection URL. If you use this flag, do not set any other connection flags.<br><br>For insecure connections, the URL format is: <br>`--url=postgresql://<user>@<host>:<port>/<database>?sslmode=disable`<br><br>For secure connections, the URL format is:<br>`--url=postgresql://<user>@<host>:<port>/<database>`<br>with the following parameters in the query string:<br>`sslcert=<path-to-client-crt>`<br>`sslkey=<path-to-client-key>`<br>`sslmode=verify-full`<br>`sslrootcert=<path-to-ca-crt>` <br><br>**Env Variable:** `COCKROACH_URL`
> `--user`<br>`-u` | The user executing the `dump` command. The user must have the `SELECT` privilege on the target table. 
> ```

Here it's "must have the SELECT privilege", above it's "must have SELECT privileges". Should we make these consistent?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/512#-:-KNfPLZ8GN444CeIvcy-:1417088721)*
<!-- Sent from Reviewable.io -->

jseldess commented:
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[back-up-and-restore-data.md, line 47 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/512#-KNfP6tkJf9Xz2VgW_3u:-KNgk27CWK57ZRLlU4nR:-1757083253) ([raw file](https://github.com/cockroachdb/docs/blob/966b6728cc133a789080a42f8e0aa215f6d50e39/back-up-and-restore-data.md#L47)):*
<details><summary><i>Previously, mjibson (Matt Jibson) wrote…</i></summary>
> Here it's "must have the SELECT privilege", above it's "must have SELECT privileges". Should we make these consistent?
</details>
Yes. Thanks for catching that. I fixed that, added the default for the `user` flag, and added an example showing dump failure. Please take one more look.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/512)*
<!-- Sent from Reviewable.io -->

mjibson commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/512#-:-KNhMIrwBr6VxJWae4Mz:1417088721)*
<!-- Sent from Reviewable.io -->
